packages/dashboard/src/dashboard-actions/annotations.ts

```diff
@@ -0,0 +1,89 @@
+import { XAnnotation } from '@synchro-charts/core';
+import { DashboardConfiguration } from '../types';
+
+export const addXAnnotation = ({
+  dashboardConfiguration,
+  widgetId,
+  annotation,
+}: {
+  dashboardConfiguration: DashboardConfiguration;
+  widgetId: string;
+  annotation: XAnnotation;
+}) => {
+  return dashboardConfiguration.map((widget) => {
+    if (widget.id == widgetId) {
+      let currAnnotations = widget.annotations;
+      if (currAnnotations && currAnnotations.x) {
+        currAnnotations.x.push(annotation);
+      } else if (!currAnnotations) {
+        currAnnotations = {
+          x: [annotation],
+        };
+      } else {
+        currAnnotations = {
+          ...currAnnotations,
+          x: [annotation],
+        };
```

Comment:
We want t avoid this conditional approach, instead just state:

```
const currAnnotations = widget.Annotations || {}
const currXAnnotations = currAnnotations.x || []
return { ...widget, annotations: { ...currAnnotations, x: [...currXAnnotations, annotation] ]}
```

And avoid all the mutating and if blocks

Reply:
Refactored to fit this scheme